### ISSUE TITLE
False positive with EmptyForStatement check

### ISSUE THREAD

**Issue by raynebc** [oclint/oclint#354]
I am new to oclint, but while reviewing some initial results I find that this check flags instances of code where for loops perform actions in addition to manipulating a counter.  For example:
```
int ctr;
char array[10];
for(ctr = 0; ctr < 10; array[ctr] = ctr, ctr++);
```
This isn't necessarily the prettiest code, but it is more compact than adding another line and is perfectly valid in C.

**Comment by ryuichis** [comment]
Since the for statement body is indeed empty, the rule does what it supposes to do. However, I totally understand what you mean by that. In this case, you can disable that rule entirely when you run oclint, or you can suppress it locally with `//!OCLINT`.

**Comment by raynebc** [comment]
The only reason I brought this up is because the documentation defines this check as testing for a for loop that does **nothing**, not simply one with a missing or empty body.  In this case the documentation could be clarified to reflect that it will flag loops that actually perform commands but still have an empty body.

**Comment by ryuichis** [comment]
That's fair. A pull request is highly appreciated.

**Comment by raynebc** [comment]
I haven't made one before.  It's seems somewhat involved to check out the entire project and propose a commit instead of ask if you could add something like the following to the documentation for this check:
```
Only the body of the for loop, and not the for line itself, is examined.
```
It's possible other loop types are treated the same way by the checkers, but I haven't tested for it.
